File: leetcode/longestPrefix.js
function sol(strs) {
  let answer = '';
  let temp = [];
  let min = Math.min(...strs.map(v => v.length));

  for (let i = 0; i < min; i++) {
    for (let j = 0; j < strs.length; j++) {
      temp.push(strs[j][i]);
    }
    temp = temp.reduce((acc, v) => {
      return acc == v && v;
    }, temp[0]);
    if (!temp) break;
    answer += temp;
    temp = [];
  }
  return answer;
}
console.log(sol(['flow', 'flower', 'flight']));

File: leetcode/longestPrefix.py
from functools import reduce

def sol(strs):
    min_value =len(min(strs))
    answer=''
    temp=[]
    for i in range(min_value):
        for j in range(len(strs)):
            temp.append(strs[j][i])
        temp=reduce(lambda a,b: a==b and a,temp)
        if(temp):
            answer += temp
            temp=[]
        else:
            break;
    return answer

print(sol(['rdog','racecar','rcar']))
